src/main/java/org/schemaspy/view/MustacheCompiler.java

```diff
@@ -1,99 +1,10 @@
 
-import com.github.mustachejava.DefaultMustacheFactory;
-import com.github.mustachejava.Mustache;
-import com.github.mustachejava.MustacheFactory;
-import org.schemaspy.util.DataTableConfig;
-
 import java.io.IOException;
-import java.io.StringWriter;
 import java.io.Writer;
-import java.util.HashMap;
-import java.util.function.Function;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
-
-/**
- * Created by rkasa on 2016-03-22.
- *
- * @author Rafal Kasa
- * @author Ismail Simsek
- * @author Thomas Traude
- * @author Daniel Watt
- * @author Nils Petzaell
- */
-public class MustacheCompiler {
-
-    private final String databaseName;
-    private final HtmlConfig htmlConfig;
-    private final DataTableConfig dataTableConfig;
-    private final MustacheFactory mustacheFactory;
-
-    public MustacheCompiler(String databaseName, HtmlConfig htmlConfig, DataTableConfig dataTableConfig) {
-        this.databaseName = databaseName;
-        this.htmlConfig = htmlConfig;
-        this.dataTableConfig = dataTableConfig;
-        this.mustacheFactory = new DefaultMustacheFactory(new MustacheCustomResolver(htmlConfig.getTemplateDirectory()));
-    }
-
-    public void write(PageData pageData, Writer writer) throws IOException {
-        StringWriter result = new StringWriter();
-
-        HashMap<String, Object> pageScope = new HashMap<>();
-        pageScope.put("toFileName", (Function<String,String>) new FileNameGenerator()::generate);
-        pageScope.put("databaseName", databaseName);
-        pageScope.put("paginationEnabled", htmlConfig.isPaginationEnabled());
-        pageScope.put("displayNumRows", htmlConfig.isNumRowsEnabled());
-        pageScope.put("dataTableConfig", dataTableConfig.getPageScopeMap());
-        pageScope.putAll(pageData.getScope());
-
-        Mustache mustachePage = mustacheFactory.compile(pageData.getTemplateName());
-        mustachePage.execute(result, pageScope).flush();
-
-        HashMap<String, Object> containerScope = new HashMap<>();
-        containerScope.put("content", result);
-        containerScope.put("pageScript", pageData.getScriptName());
-        containerScope.put("rootPath", getRootPath(pageData.getDepth()));
-        containerScope.put("rootPathtoHome", getRootPathToHome(pageData.getDepth()));
-        containerScope.putAll(pageScope);
-
-        Mustache mustacheContainer = mustacheFactory.compile("container.html");
-        mustacheContainer.execute(writer, containerScope).flush();
-    }
-
-
-    public String getRootPath(int depth) {
-        return IntStream.range(0, depth).mapToObj(i -> "../").collect(Collectors.joining("", "", ""));
-    }
 
-    private String getRootPathToHome(int depth) {
-        String path = getRootPath(depth);
-        if (htmlConfig.isOneOfMultipleSchemas()) {
-            path += "../";
-        }
-        return path;
-    }
+public interface MustacheCompiler {
+    void write(PageData pageData, Writer writer) throws IOException;
```

Comment:
Consider moving parameters to constructor (if possible)

Reply:
Starts becoming scope creep.

Since it would invert all of the Html* classes in the view package.

It should be possible, since we don't have that many templates and don't create to many pages. So the caching/compiling performance isn't important.

Do you want to create an issue to look into it?

Reply:
Fair.

> Do you want to create an issue to look into it?

I don't have a strong opinion. Might just as well be administrative overhead